### PR TITLE
Feature hc sr04

### PR DESCRIPTION
**Pull request by Gian1080**
Functional code, return a reliable and accurate distance

### CODE REVIEW

File: code/headers/lidar.hpp [hunk 1]
```diff
@@ -1,118 +0,0 @@
-#pragma once
```
Comment (ThePatman3) [inline code/headers/lidar.hpp]: Why is this file deleted?

File: code/headers/lidar_packets.hpp [hunk 2]
```diff
@@ -1,45 +0,0 @@
-#pragma once
```
Comment (ThePatman3) [inline code/headers/lidar_packets.hpp]: Why is this file deleted?

File: code/src/lidar.cpp [hunk 3]
```diff
@@ -1,89 +0,0 @@
-#include <lidar.hpp>
```
Comment (ThePatman3) [inline code/src/lidar.cpp]: Why is this file deleted?

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#include <hwlib.hpp>
```
Comment (ThePatman3) [inline main.cpp]: There shouldn't be a main.cpp located in this folder.

File: code/src/HC_SR04_c.cpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+int16_t R2D2::Distance::HC_SR04_c::get_distance(){
+    trigger_pin.write(0);
+    hwlib::wait_us(2);
+    trigger_pin.write(1);
+    hwlib::wait_us(10);
+    trigger_pin.write(0);
+
+    while(!echo_pin.read()){
+        hwlib::wait_us(10);
```
Comment (ThePatman3) [inline code/src/HC_SR04_c.cpp]: I am not sure if this wait serves a purpose: without rtos, the hwlib::wait functions are just while loops waiting for the given time to pass. I think the measurements would be more accurate without this function call.

File: code/src/HC_SR04_c.cpp [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    uint32_t ticks_start = hwlib::now_ticks();
+
+    while(echo_pin.read()){
+        hwlib::wait_us(10);
```
Comment (ThePatman3) [inline code/src/HC_SR04_c.cpp]: Same here, I think the measurements would be more more accurate without the hwlib::wait.

**Review by ThePatman3 — CHANGES_REQUESTED** [review]
The HC_SR04 code looks good! I have left a few comments, mainly concerning the deletion of some files, that have to be resolved before I approve this pull request.

File: code/headers/lidar.hpp (hunk 1 above)
Comment (Gian1080) [reply code/headers/lidar.hpp]: This is in lidar.hpp in the feature branch of the sonar module, not sure how this happened

File: code/headers/lidar_packets.hpp (hunk 2 above)
Comment (Gian1080) [reply code/headers/lidar_packets.hpp]: it shouldn't be in my feature branch

File: code/src/lidar.cpp (hunk 3 above)
Comment (Gian1080) [reply code/src/lidar.cpp]: lidar, shouldn't be here

File: code/src/HC_SR04_c.cpp (hunk 5 above)
Comment (Gian1080) [reply code/src/HC_SR04_c.cpp]: This wait is made for the future RTOS, nothing else

File: code/src/HC_SR04_c.cpp (hunk 6 above)
Comment (Gian1080) [reply code/src/HC_SR04_c.cpp]: I have tested the accuracy it works well and the wait is for a future RTOS.

File: main.cpp (hunk 4 above)
Comment (Gian1080) [reply main.cpp]: You are correct, I have removed it.

File: code/headers/lidar.hpp (hunk 1 above)
Comment (Gian1080) [reply code/headers/lidar.hpp]: I returned the file and corrected the outdated usart.

File: code/headers/lidar_packets.hpp (hunk 2 above)
Comment (Gian1080) [reply code/headers/lidar_packets.hpp]: I returned the file and corrected the outdated usart.

File: code/src/lidar.cpp (hunk 3 above)
Comment (Gian1080) [reply code/src/lidar.cpp]: I returned the file and corrected the outdated usart.

**Review by ThePatman3 — APPROVED** [review]
Code looks good now and is ready to be merged.

**Comment by Gian1080** [comment]
Error can be ignored, Gerrit told me to, because Travis is bugging.
